File: Benchmarks/Core/Systems/ScriptSystem.cs
using Benchmarks.Core.Components;
using Benchmarks.Core.Entity;
using Benchmarks.Core.Scene;
using Benchmarks.Core.Systems.Contract;
using Benchmarks.Core.Types;

namespace Benchmarks.Core.Systems
{
    /// <summary>
    /// Sistema encargado de manejar los scripts de las entidades.
    /// Stub que replica el patron de iteracion del engine.
    /// </summary>
    public class ScriptSystem : ISystem
    {
        public override void InitializeSystem(EntityManager @entities)
        {
            EntityManager = @entities;
        }

        public override void SharedEntityInitialize(ref Yotsuba Entidad)
        {
            if (!Entidad.HasComponent(YTBComponent.Script)) return;
            ref ScriptComponent script = ref GetScriptComponent(Entidad.Id);
            _ = script.IsActive;
        }

        public override void SharedEntityForEachUpdate(ref Yotsuba Entidad, GameTime time)
        {
            if (!Entidad.HasComponent(YTBComponent.Script)) return;
            ref ScriptComponent script = ref GetScriptComponent(Entidad.Id);
            if (!script.IsActive) return;
            // Simulate script execution: read transform, modify it
            if (Entidad.HasComponent(YTBComponent.Transform))
            {
                ref TransformComponent transform = ref GetTransform2DComponent(Entidad.Id);
                _ = transform.Position;
            }
        }

        public void DrawSystem2D(GameTime gameTime)
        {
            if (EntityManager == null) return;
            foreach (ref Yotsuba entity in GetEntitiesAsSpan())
            {
                if (!entity.HasComponent(YTBComponent.Script)) continue;
                ref ScriptComponent script = ref GetScriptComponent(entity.Id);
                if (!script.IsActive) continue;
                _ = script.ScriptCount;
            }
        }

        public void DrawSystem3D(GameTime gameTime)
        {
            // Stub for 3D script rendering
        }

        public void Clear() { }
        public override void Dispose() { }
    }
}

File: Benchmarks_FSharp/Core/Systems/ScriptSystem.fs
namespace Benchmarks.Core.Systems

open Benchmarks.Core.Components
open Benchmarks.Core.Entity
open Benchmarks.Core.Scene
open Benchmarks.Core.Systems.Contract
open Benchmarks.Core.Types

type ScriptSystem() =
    inherit ISystem()

    override this.InitializeSystem(entities: EntityManager) =
        this.EntityManager <- entities

    override this.SharedEntityInitialize(entidad: byref<Yotsuba>) =
        if not (entidad.HasComponent(YTBComponent.Script)) then ()
        else
        let script = &this.GetScriptComponent(entidad.Id)
        script.IsActive |> ignore

    override this.SharedEntityForEachUpdate(entidad: byref<Yotsuba>, time: GameTime) =
        if not (entidad.HasComponent(YTBComponent.Script)) then ()
        else
        let script = &this.GetScriptComponent(entidad.Id)
        if not script.IsActive then ()
        else
        if entidad.HasComponent(YTBComponent.Transform) then
            let transform = &this.GetTransform2DComponent(entidad.Id)
            transform.Position |> ignore

    member this.DrawSystem2D(gameTime: GameTime) =
        if isNull this.EntityManager then ()
        else
        let entities = this.GetEntitiesAsSpan()
        for i = 0 to entities.Length - 1 do
            let entity = &entities.[i]
            if entity.HasComponent(YTBComponent.Script) then
                let script = &this.GetScriptComponent(entity.Id)
                if script.IsActive then
                    script.ScriptCount |> ignore

    member _.DrawSystem3D(gameTime: GameTime) = ()

    member _.Clear() = ()
    override _.Dispose() = ()
